Lib/tarfile.py

```diff
@@ -1669,21 +1670,21 @@ def gzopen(cls, name, mode="r", fileobj=None, compresslevel=9, **kwargs):
         try:
             from gzip import GzipFile
         except ImportError:
-            raise CompressionError("gzip module is not available")
+            raise CompressionError("gzip module is not available") from None
 
         try:
             fileobj = GzipFile(name, mode + "b", compresslevel, fileobj)
-        except OSError:
+        except OSError as e:
             if fileobj is not None and mode == 'r':
-                raise ReadError("not a gzip file")
+                raise ReadError("not a gzip file") from e
```

Comment:
Why set `__cause__`? The initial exception is already linked as `__context__`.

Reply:
So the traceback correctly identifies that [the first exception is the cause](https://bugs.python.org/issue39717?@ok_message=issue%2039717%20versions%20edited%20ok&@template=item#msg362444), and the second (current) exception is not a bug in the exception handler.